library/src/main/java/com/getstream/sdk/chat/viewmodel/channels/ChannelsViewModel.kt

```diff
@@ -83,6 +80,34 @@ public class ChannelsViewModelImpl(
         }.exhaustive
     }
 
+    private fun loadChannels() {
+        queryChannels().run {
+            channelsData = map(channels) { it.toState() }
+            loadingMoreData = map(loadingMore, ChannelsViewModel.State::LoadingNextPage)
+            endPageData = map(endOfChannels, ChannelsViewModel.State::EndPageReached)
+        }
+
+        stateMerger.addSource(channelsData, stateMerger::setValue)
+        stateMerger.addSource(loadingMoreData, stateMerger::setValue)
```

Comment:
Not relevant as the channel list should be updated when on it's own, but in such cases we need not to forget to remove the old source prior to adding a replacement, e.g.:

```
threadListData?.let {
            stateMerger.apply {
                removeSource(messageListData)
                addSource(it) { value = State.Result(it) }
            }
        }
```